packages/authorization-process/src/services/validators.ts

```diff
@@ -176,20 +178,32 @@ export const assertSecurityRoleIsClientMember = (
   }
 };
 
-export const assertClientIsConsumer = (client: Client): void => {
+export function assertClientIsConsumer(
+  client: Client
+): asserts client is Client & { kind: typeof clientKind.consumer } {
   if (client.kind !== clientKind.consumer) {
     throw clientKindNotAllowed(client.id);
   }
-};
+}
 
-export const assertClientIsAPI = (client: Client): void => {
+export function assertClientIsAPI(
+  client: Client
+): asserts client is Client & { kind: typeof clientKind.api } {
   if (client.kind !== clientKind.api) {
     throw clientKindNotAllowed(client.id);
   }
-};
+}
 
 export const assertAdminInClient = (client: Client, adminId: UserId): void => {
   if (client.adminId !== adminId) {
     throw clientAdminIdNotFound(client.id, adminId);
   }
 };
+
+export function assertJwkKtyIsDefined(
+  jwk: JsonWebKey
+): asserts jwk is JsonWebKey & { kty: NonNullable<JsonWebKey["kty"]> } {
+  if (jwk.kty === undefined) {
+    throw genericError("JWK must have a 'kty' property");
```

Comment:
```suggestion
    throw invalidClaim("JWK must have a 'kty' property");
```
I would use a more specific error, maybe we can reuse this one?